text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/placeholder/PlaceholderResolver.java

```diff
@@ -194,29 +126,20 @@
   /**
    * Constructs a placeholder resolver capable of dynamically resolving placeholders.
    *
-   * <p>The {@code resolver} function must return instances of String, {@link ComponentLike} or {@link Placeholder}.
-   * The resolver can return {@code null} to indicate it cannot resolve a placeholder.</p>
+   * <p>
+   *   The resolver can return {@code null} to indicate it cannot resolve a placeholder.
+   *   Once a string to replacement mapping has been created, it will be cached to avoid
+   *   the cost of recreating the replacement.
+   * </p>
```

Comment:
```suggestion
   * <p>The resolver can return {@code null} to indicate it cannot resolve a placeholder.
   *   Once a string to replacement mapping has been created, it will be cached to avoid
   *   the cost of recreating the replacement.</p>
```